packages/nx/schemas/nx-schema.json

```diff
@@ -0,0 +1,127 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://nx.dev/core-concepts/configuration#nxjson",
+  "title": "JSON schema for Nx configuration",
+  "type": "object",
+  "properties": {
+    "implicitDependencies": {
+      "type": "object",
+      "description": "Map of files to projects that implicitly depend on them."
+    },
+    "affected": {
+      "type": "object",
+      "description": "Default options for `nx affected`.",
+      "properties": {
+        "defaultBase": {
+          "type": "string",
+          "description": "Default based branch used by affected commands."
+        }
+      },
+      "additionalProperties": false
+    },
+    "npmScope": {
+      "type": "string",
+      "description": "NPM Scope that the workspace uses."
+    },
+    "tasksRunnerOptions": {
+      "additionalProperties": {
+        "$ref": "#/definitions/tasksRunnerOptions"
+      }
+    },
+    "targetDependencies": {
+      "type": "object",
+      "description": "Dependencies between different target names across all projects.",
+      "additionalProperties": {
+        "$ref": "#/definitions/targetDependencyConfig"
+      }
+    },
+    "workspaceLayout": {
+      "type": "object",
+      "description": "Where new apps + libs should be placed.",
+      "properties": {
+        "libsDir": {
+          "type": "string",
+          "description": "Default folder name for libs."
+        },
+        "appsDir": {
+          "type": "string",
+          "description": "Default folder name for apps."
+        }
+      },
+      "additionalProperties": false
+    },
+    "cli": {
+      "$ref": "#/definitions/cliOptions"
+    },
+    "generators": {
+      "$ref": "#/definitions/generatorOptions"
+    },
+    "plugins": {
+      "type": "array",
+      "description": "Plugins for extending the project graph.",
+      "items": {
+        "type": "string"
+      }
+    },
+    "defaultProject": {
+      "type": "string",
+      "description": "Default project. When project isn't provided, the default project will be used."
+    }
+  },
+  "definitions": {
+    "cliOptions": {
+      "type": "object",
+      "description": "Default generator collection.",
+      "properties": {
+        "packageManager": {
+          "type": "string",
+          "description": "The default package manager to use.",
+          "enum": ["yarn", "pnpm", "npm"]
+        },
+        "defaultCollection": {
+          "type": "string",
+          "description": "The default schematics collection to use."
+        }
+      },
+      "additionalProperties": false
```

Comment:
Some other angular cli props can be added here, so we should probably leave this off